app/src/components/ChatBot/chat-popover.tsx

```diff
@@ -35,27 +29,24 @@ export default function ChatPopover({
   const inputRef = React.useRef(null);
   const { t } = useTranslation(lng, "chat");
 
+  const onOpenChange = (e: OpenChangeDetails) => {
+    if (!e.open) {
+      onClose();
+    } else {
+      onOpen();
+    }
+  };
+
   return (
     <>
       <PopoverRoot
         open={open}
         initialFocusEl={() => inputRef.current}
-        onOpenChange={onOpen}
+        onOpenChange={onOpenChange}
+        onExitComplete={onClose}
```

Comment:
If we have the onOpenChange listener, do we need onExitComplete as well?
Or does it not trigger in certain situations otherwise?